src/XrdHttpTpc/XrdHttpTpcState.cc

```diff
@@ -159,13 +162,51 @@ void State::SetupHeaders(XrdHttpExtReq &req) {
     }
 }
 
+void State::SetupHeadersForHEAD(XrdHttpExtReq &req) {
+  struct curl_slist *list = NULL;
+  for (std::map<std::string, std::string>::const_iterator hdr_iter = req.headers.begin();
+       hdr_iter != req.headers.end();
+       hdr_iter++) {
+    if (!strcasecmp(hdr_iter->first.c_str(),"copy-header")) {
+      list = curl_slist_append(list, hdr_iter->second.c_str());
+    }
+    // Note: len("TransferHeader") == 14
+    if (!strncasecmp(hdr_iter->first.c_str(),"transferheader",14)) {
+      std::stringstream ss;
+      ss << hdr_iter->first.substr(14) << ": " << hdr_iter->second;
+      list = curl_slist_append(list, ss.str().c_str());
+    }
+  }
+  if(!req.mReprDigest.empty()) {
+    size_t reprDigestSize = req.mReprDigest.size();
+    std::stringstream ss;
+    ss << "Want-Repr-Digest: ";
+    size_t cpt = 1;
+    for (const auto &kv: req.mReprDigest) {
+      // We put the same weight for the digest names as we do not have any way, according to the specs,
+      // to give priority to a digest name in particular
+      ss << kv.first << '=' << 5;
+      if(cpt < reprDigestSize) {
+        ss << ',';
+      }
+      cpt++;
+    }
+    list = curl_slist_append(list, ss.str().c_str());
+  }
+
+  if (list != NULL) {
```

Comment:
Minor C++ style, not needed to change, but this is better as `if (list)` or `if (list != nullptr)`.